.github/workflows/upgrade-test.yml

```diff
@@ -0,0 +1,189 @@
+        env:
+          BINARY_ROOT_PATH: ./latest-release-bins
+        run: |
+          set -x
+          mkdir -p /tmp/chainflip/bashful
+          mkdir -p /tmp/chainflip/doc
+          mkdir -p /tmp/chainflip/dopey
+          touch ./localnet/.setup_complete
+          ./localnet/manage.sh
+
+      - name: Run bouncer on latest release
+        id: pre-upgrade-bouncer
+        working-directory: bouncer
+        run: |
+          ./run.sh
+
+      # we need to be sure that when this fails, we catch the error, any panics etc. that occur
+      # TODO: Run swaps simultaneously to the upgrade - we could do that inside the `upgrade_network` command itself.
+      - name: Upgrade network
+        shell: bash
+        id: upgrade-network
+        working-directory: bouncer
+        run: |
+          ./commands/upgrade_network.ts prebuilt \
+            --runtime ./../main-runtime/state_chain_runtime.compact.compressed.wasm \
+            --bins ./../main-bins \
+            --localnet_init ./../localnet/init \
+            --oldVersion "${{ env.RELEASE_VERSION }}"
+
+      - name: Run bouncer after upgrade
+        id: post-upgrade-bouncer
+        working-directory: bouncer
+        run: |
+          ./run-no-setup.sh
+
+      - name: Print chainflip-engine logs
+        if: failure()
+        run: |
+          cat /tmp/chainflip/*/chainflip-engine.log
+
+      - name: Print chainflip-node logs
+        if: failure()
+        run: |
+          cat /tmp/chainflip/*/chainflip-node.log
+
+      - name: Print broker logs
+        if: failure()
+        run: |
+          cat /tmp/chainflip/chainflip-broker-api.log
+
+      - name: Print lp-api logs
+        if: failure()
+        run: |
+          cat /tmp/chainflip/chainflip-lp-api.log
+
+      - name: Upload Localnet Logs 💾
+        if: always()
+        continue-on-error: true
+        uses: actions/upload-artifact@v3
+        with:
+          name: localnet-logs
+          path: |
+            /tmp/chainflip/*/chainflip-*.log
+
+      - name: Clean Up docker containers 🧹
+        if: always()
+        continue-on-error: true
+        run: |
+          ls -alR /tmp/chainflip
+          docker compose -f localnet/docker-compose.yml -p "chainflip-localnet" logs
+          docker compose -f localnet/docker-compose.yml -p "chainflip-localnet" down --rmi all --volumes --remove-orphans
+
+      - name: Notify on failed bouncer
+        if: failure() && github.ref_name == 'main' || cancelled() && github.ref_name == 'main'
+        env:
+          DISCORD_USERNAME: "HEUTE LEIDER NICHT"
+          DISCORD_WEBHOOK: ${{ secrets.CF_DISCORD_ALERTS_CRITICAL }}
+        uses: Ilshidur/action-discord@0.3.2
+        with:
+          args: |
+            ❗️❗️❗️❗️ Sorry **${{ github.actor }}**, the Bouncer has rejected you ❗️❗️❗️❗️
+            👾 Link to job: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+            #️⃣ Tagging: <@&939151218708709416>
```

Comment:
Maybe change this to be more descriptive of the what is failing i.e. the upgrade test